### PR TITLE
Use Vuetify Themes

### LINKED ISSUE #714: Dark Mode Support
**What would you like to be added**:
It would be pretty cool if we would have a dark mode support in the Gardener dashboard -  potentially adapting the OS preferences or a switch button somewhere on the dashboard.

**Why is this needed**:
Coz dark mode is dope :-)

### LINKED ISSUE #873: Make colors configurable
**What would you like to be added**:
Make colors of the Dashboard configurable.
To achieve this, a first step is to remove all hard coded color values and color names (such as `cyan darken-2` or `red`) and use `primary` or `error` color variables instead. These variables can be centrally overwritten by configuration.

**Why is this needed**
This is needed to adapt the styling of the Dashboard to various environments / brandings.

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
This PR is a big step to make the dashboard customizable. By using vuetify themes it is possible to centrally configure the colors used by the dashboard. This also allows us to introduce dark mode.

**Which issue(s) this PR fixes**:
Fixes #873 Fixes #714

**Special notes for your reviewer**:

**Release note**:

```feature developer
Use Vuetify Themes to define Dashboard color scheme. This makes it easier to change colors and adaptions will not be overwritten by future code changes
```
```feature user
Added Dark Mode support
```
```feature user
Redesigned, cleaner Login Page
```

### CODE REVIEW

File: frontend/src/app.js [hunk 1]
```diff
@@ -33,6 +33,9 @@ function createApp (vuetify) {
     router: createRouter(store),
     render (createElement) {
       return createElement(App)
+    },
+    beforeCreate () {
+      this.$store.dispatch('initStore')
```
Comment (holgerkoser) [inline frontend/src/app.js:38]:
```suggestion
```

File: frontend/src/components/MainToolbar.vue [hunk 2]
```diff
@@ -95,12 +95,18 @@ SPDX-License-Identifier: Apache-2.0
...
-            <v-btn block text color="cyan darken-2" class="justify-start" :to="accountLink" title="My Account">
+            <v-btn block text color="primary" class="justify-start" :to="accountLink" title="My Account">
               <v-icon class="mr-3">mdi-account-circle</v-icon>
               My Account
             </v-btn>
           </v-card-actions>
           <v-divider></v-divider>
+          <v-card-actions class="px-3">
+            <v-icon color="primary" class="ml-2 mr-3">mdi-brightness-6</v-icon>
+            <v-switch v-model="darkModeValue" dense hide-details class="ma-0" color="primary" @click.native.stop></v-switch>
```
Comment (holgerkoser) [inline frontend/src/components/MainToolbar.vue:106]:
```suggestion
            <v-switch v-model="darkMode" dense hide-details class="ma-0" color="primary" @click.native.stop></v-switch>
```

File: frontend/src/components/MainToolbar.vue [hunk 3]
```diff
@@ -142,7 +148,8 @@ export default {
   methods: {
     ...mapActions([
       'setSidebar',
-      'setError'
+      'setError',
+      'setDarkMode'
```
Comment (holgerkoser) [inline frontend/src/components/MainToolbar.vue:152]:
```suggestion
      'setError'
```

File: frontend/src/app.js (hunk 1 above)
Comment (grolu) [reply frontend/src/app.js:38]: done

File: frontend/src/app.js [hunk 4]
```diff
@@ -20,22 +20,23 @@ const App = Vue.extend({
         this.$bus.$emit('esc-pressed')
       }
     })
+
+    const darkMode = Vue.localStorage.getItem('global/dark-mode') === 'true'
```
Comment (holgerkoser) [inline frontend/src/app.js:24]:
```suggestion
    const darkMode = this.$localStorage.getItem('global/dark-mode') === 'true'
```

File: frontend/src/store/index.js [hunk 5]
```diff
@@ -1006,6 +1003,9 @@ const getters = {
   },
   onlyShootsWithIssues (state, getters) {
     return getters['shoots/onlyShootsWithIssues']
+  },
+  getDarkMode (state) {
+    return state.darkMode
```
Comment (holgerkoser) [inline frontend/src/store/index.js:1008]:
```suggestion
```

File: frontend/src/components/MainToolbar.vue [hunk 6]
```diff
@@ -198,9 +198,9 @@ export default {
         query
       }
     },
-    darkModeValue: {
+    darkMode: {
       get () {
-        return this.darkMode
+        return this.getDarkMode
```
Comment (holgerkoser) [inline frontend/src/components/MainToolbar.vue:203]:
```suggestion
        return this.$store.state.darkMode
```

File: frontend/src/components/MainToolbar.vue [hunk 7]
```diff
@@ -161,14 +161,14 @@ export default {
...
-      'darkMode'
+      'cfg'
     ]),
     ...mapGetters([
       'username',
       'displayName',
       'avatarUrl',
-      'isAdmin'
+      'isAdmin',
+      'getDarkMode'
```
Comment (holgerkoser) [inline frontend/src/components/MainToolbar.vue:171]:
```suggestion
```

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Comment by valentinvieriu** [comment]
Dark mode looks awesome!!!!! Can't wait to enable it!

**Comment by rfranzke** [comment]
Is this synced automatically based on the system settings on MacOS? :)

**Comment by grolu** [comment]
>Is this synced automatically based on the system settings on MacOS? :)

Currently not. Maybe we can think about something like this when we introduce user settings.

**Comment by rfranzke** [comment]
Yeah, thanks @grolu, would be great!

**Pull request merged**